Source/SwiftLintFramework/Rules/QuickDiscouragedCallRule.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  QuickDiscouragedCallRule.swift
+//  SwiftLint
+//
+//  Created by Ornithologist Coder on 8/11/17.
+//  Copyright © 2017 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+// swiftlint:disable identifier_name
+
+public struct QuickDiscouragedCallRule: ASTRule, OptInRule, ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "quick_discouraged_call",
+        name: "Quick Discouraged Call",
+        description: "Discouraged call inside 'describe' and/or 'context' block.",
+        kind: .style,
+        nonTriggeringExamples: QuickDiscouragedCallRuleExamples.nonTriggeringExamples,
+        triggeringExamples: QuickDiscouragedCallRuleExamples.triggeringExamples
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        guard
+            fileContainsQuickSpec(file: file),
```

Comment:
Err, actually, I think we have a larger issue here. I believe that violations will be generated more than one time, because `ASTRule` navigates the structure and this rule does the same in `violationOffsets(in:)`.

Maybe it'd be better to change it to be a `Rule` and fix both issues. You can even validate if you're inside an `override func spec()` (a instance method, with `override` annotation and no parameters).